jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushBlobStep.java

```diff
@@ -77,14 +77,24 @@ public BlobDescriptor call() throws IOException, RegistryException, ExecutionExc
               .newRegistryClient();
       registryClient.setTimer(timer);
 
+      // check if the BLOB is available
       if (registryClient.checkBlob(blobDescriptor.getDigest()) != null) {
         buildConfiguration
             .getBuildLogger()
             .info("BLOB : " + blobDescriptor + " already exists on registry");
         return blobDescriptor;
       }
 
-      registryClient.pushBlob(blobDescriptor.getDigest(), blob);
+      // If base and target images are in the same registry, then use mount/from to try mounting the
+      // BLOB from the base image repository to the target image repository and possibly avoid
+      // having to push the BLOB. See
+      // https://docs.docker.com/registry/spec/api/#cross-repository-blob-mount for details.
+      boolean sameRegistry =
+          buildConfiguration
+              .getBaseImageRegistry()
+              .equals(buildConfiguration.getTargetImageRegistry());
```

Comment:
Btw, the presubmit is failing because I changed `BuildConfiguration` yesterday. You'll need to do something like `buildConfiguration.getTargetImageConfiguration().getImageRegistry()` now.